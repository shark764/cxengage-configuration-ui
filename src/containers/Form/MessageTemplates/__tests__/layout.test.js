/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import MessageTemplatesForm from '../layout';

describe('<MessageTemplatesForm />', () => {
  it('renders create form', () => {
    const rendered = shallow(
      <MessageTemplatesForm
        isSaving={false}
        selectedEntityId="create"
        initialValues="mockInitialValues"
        userHasUpdatePermission={true}
        channels={List(['mockChannels'])}
        templateText="mockTemplateText"
        templateTextType="plaintext"
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update form', () => {
    const rendered = shallow(
      <MessageTemplatesForm
        isSaving={false}
        selectedEntityId="mockEntityId"
        initialValues="mockInitialValues"
        userHasUpdatePermission={true}
        channels={List(['mockChannels'])}
        templateText="mockTemplateText"
        templateTextType="html"
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
