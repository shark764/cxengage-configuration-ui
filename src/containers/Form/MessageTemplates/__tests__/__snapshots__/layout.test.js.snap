// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageTemplatesForm /> renders create form 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    componentType="input"
    data-automation="nameInput"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    data-automation="descriptionInput"
    disabled={false}
    inputType="text"
    label="Description"
    name="description"
  />
  <CheckboxField
    buttonType="checkboxField"
    data-automation="channelsDropdown"
    disabled={false}
    dropDownText="Select Channel Type..."
    items={
      Array [
        Object {
          "label": "SMS",
          "name": "sms",
        },
        Object {
          "label": "Messaging",
          "name": "messaging",
        },
        Object {
          "label": "Email",
          "name": "email",
        },
      ]
    }
    label="Channels *"
    name="channels"
  />
  <SelectField
    data-automation="typeList"
    disabled={false}
    label="Type *"
    name="templateTextType"
    options={
      Array [
        Object {
          "label": "Plain Text",
          "value": "plaintext",
        },
        Object {
          "label": "Rich Text",
          "value": "html",
        },
      ]
    }
    required={true}
  />
  <InputField
    componentType="textarea"
    data-automation="templateInput"
    disabled={false}
    inputType="text"
    label="Template *"
    name="template"
  />
</form>
`;

exports[`<MessageTemplatesForm /> renders update form 1`] = `
<form
  onSubmit={[Function]}
>
  <InputField
    componentType="input"
    data-automation="nameInput"
    disabled={false}
    inputType="text"
    label="Name *"
    name="name"
  />
  <InputField
    componentType="textarea"
    data-automation="descriptionInput"
    disabled={false}
    inputType="text"
    label="Description"
    name="description"
  />
  <CheckboxField
    buttonType="checkboxField"
    data-automation="channelsDropdown"
    disabled={true}
    dropDownText="Select Channel Type..."
    items={
      Array [
        Object {
          "label": "SMS",
          "name": "sms",
        },
        Object {
          "label": "Messaging",
          "name": "messaging",
        },
        Object {
          "label": "Email",
          "name": "email",
        },
      ]
    }
    label="Channels *"
    name="channels"
  />
  <SelectField
    data-automation="typeList"
    disabled={false}
    label="Type *"
    name="templateTextType"
    options={
      Array [
        Object {
          "label": "Plain Text",
          "value": "plaintext",
        },
        Object {
          "label": "Rich Text",
          "value": "html",
        },
      ]
    }
    required={true}
  />
  <RichTextEditorField
    data-automation="templateEditor"
    disabled={false}
    label="Template *"
    name="template"
    templateText="mockTemplateText"
    toggleMessageTemplateText={[Function]}
  />
</form>
`;
